src/lib.rs

```diff
@@ -80,14 +80,29 @@ struct Reporter {
 #[pymethods]
 impl Reporter {
     #[new]
-    fn new() -> PyResult<Reporter> {
+    fn new(config: Option<&PyDict>) -> PyResult<Reporter> {
+        let mut agent_host_name: String = "127.0.0.1".to_string();
+        let mut agent_port: i32 = 6831;
+
+        if let Some(config) = config {
+            agent_host_name = config
+                .get_item("agent_host_name")
+                .unwrap()
```

Comment:
As a general rust note: `unwrap()` should really only be used for things that are programming bugs, since it will cause the app to panic. Also, in general you should prefer `.expect("some explanation")` to give a hint as to what has gone wrong if we do panic.

Reply:
https://learning-rust.github.io/docs/e4.unwrap_and_expect.html has some more details but I am still squinting at most of this stuff trying to understand.